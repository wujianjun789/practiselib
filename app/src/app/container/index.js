/**
 * Created by a on 2017/6/29.
 */
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types';
import '../../../public/styles/app.less';
import Card from './Card';
import UserCenter from '../../common/containers/UserCenter';
import Overlayer from '../../common/containers/Overlayer';
import {loginHandler} from '../../api/login'
import {getModule} from '../action/index'
/**
 * @param {String} title required
 * @param {String} name  required
 * @param {Array}  items required
 */
export class App extends Component{
    constructor(props){
        super(props);
        this.loginFail = this.loginFail.bind(this);
        this.loginSuccess = this.loginSuccess.bind(this);
    }

    componentWillMount(){
        console.log(localStorage)
        /*if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            
            this.props.router.push('/login')
            
        }else{

            this.props.actions.loginHandler(sessionStorage.username, sessionStorage.password, this.loginSuccess, this.loginFail);

        }*/
        // if(this.props.userCenter.islogin!=1){
        //     this.props.router.push('/login')
        // }
        const {actions} = this.props;
        actions && actions.getModule();
    }

    componentDidMount() {
    }

    componentWillReceiveProps(nextProps) {



    }

    loginFail(){
        this.setState({style:{visibility: 'visible'}})
    }

    loginSuccess(){
        this.props.router.push('/')
    }

    render(){
        const {title="StarRiver", name="智慧路灯管理系统", items} = this.props;
        return (
            <div className="app">
                <div className="header clearfix">
                    <div className="header-left clearfix">
                        <span className="icon icon-logo"></span>
                        <span className="tit">{title}</span>
                    </div>
                    <UserCenter router={this.props.router}/>
                    <div className="header-right clearfix">
                        <span className="name">{name}</span>
                    </div>
                </div>
                <div className="cont">
                    <ul className="clearfix">
                        {items.map((item, index)=><li key={item.key}><Card _key={item.key} title={item.title} link={item.link}/></li>)}
                    </ul>
                </div>
                <Overlayer />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    let app = state.app;
    return {
        title: app.title,
        name: app.name,
        items: app.items,
        userCenter:state.userCenter
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            getModule: getModule,
            loginHandler
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            key: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired
        })
    ).isRequired
}