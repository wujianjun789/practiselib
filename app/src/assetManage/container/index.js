/**
 * Created by a on 2017/7/17.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {injectIntl} from 'react-intl';

import '../../../public/styles/assetManage-model.less';
import '../../../public/styles/assetManage-statistics.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '../../data/assetModels';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';

import {sideBarToggled} from '../action/index';
import {treeViewNavigator} from '../../common/util/index';
class AssetManageIndex extends Component {
  constructor(props) {
    super(props);

    this.state = {
      pathname: ""
    }

    this.initModel = this.initModel.bind(this);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);

    this.getMathNode = this.getMathNode.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    this.initModel();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !==  this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }
  componentDidUpdate(){
    // console.log(this.props);
    // const {location} = this.props;
    // if(location.pathname !== this.state.pathname){
    //   this.setState({pathname: location.pathname}, ()=>{
    //     this.initModel();
    //   });
    // }
  }
  initModel(){
    getModelData(() => {this.mounted && this.initTreeData();});
  }

  initTreeData() {
    TreeData.map(tree=>{
      return Object.assign({}, tree, {name:this.props.intl.formatMessage({id:tree.name})});
    })
    this.props.actions.treeViewInit(TreeData);
    treeViewNavigator(TreeData, this.props.router);

    const node = this.getMathNode(TreeData, this.props.location.pathname);
    if(node){
      this.onToggle(node);
    }
  }

  getMathNode(data, pathname){
    for(let i=0;i<data.length;i++){
      if(!data[i].children && data[i].link.indexOf(pathname) > -1){
        return data[i];
      }else{
        if(data[i].children){
          const node = this.getMathNode(data[i].children, pathname);
          if(node){
            return node;
          }
        }

      }
    }

    return null;
  }

  onToggle(node) {
    const {actions} = this.props;
    actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }

    return <div className={'container ' + 'asset-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName="app.asset.manage" router={this.props.router}/>
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById,
      sideBarToggled: sideBarToggled,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AssetManageIndex));