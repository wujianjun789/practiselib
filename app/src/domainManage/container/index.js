/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {TreeData} from '.././../data/domainModel'
import {treeViewInit} from '../../common/actions/treeView'
import {sideBarToggled} from '../action/index'
class DomainManageIndex extends Component{
    constructor(props){
        super(props);
        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            this.props.router.push('/login') 
        }else{}
        this.initTreeData();
    }

    componentWillUnmount(){

    }

    componentDidMount(){
        console.log(this.props.userCenter); 
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData);
    }

    onToggle(node){
        this.props.actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {children} = this.props;
        if(children){
            path = children.props.route.path;
        }
        return <div className={"container "+"domain-"+path}>
            <HeadBar moduleName={/*path=="domainEdit"?*/"域管理"/*:"地图预览"*/} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainManageIndex);