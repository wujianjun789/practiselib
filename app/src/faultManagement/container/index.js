import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Table from '../../components/Table';
import Page from '../../components/Page';
import SideBarInfo from '../../components/SideBarInfo';
import Select from '../../components/Select.1';
import ConfirmPopup from '../../components/ConfirmPopup';
import Immutable from 'immutable';
import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';

import Content from '../../components/Content';
import { treeViewInit } from '../../common/actions/treeView';
import { injectIntl } from 'react-intl';
import { getChildDomainList } from '../../api/domain';
import { DatePicker } from 'antd';
import moment from 'moment';
import PieChart from '../../lightManage/utils/pieChart';
import { spliceInArray } from '../../util/algorithm';
import { getFaultOrAlertList } from '../../api/asset';

export class Fault extends Component {
  constructor(props) {
    super(props);
    this.state = {
      model: '',
      collapse: false,
      infoCollapse: false,
      page: Immutable.fromJS({
        pageSize: 15,
        current: 1,
        total: 0
      }),
      data: Immutable.fromJS([]),
      domainList: {
        titleField: 'name',
        valueField: 'name',
        index: 0,
        value: '',
        options: []
      },
      levelList: {
        titleField: 'title',
        valueField: 'name',
        index: 0,
        value: 'deadly',
        options: [
          {
            title: '致命',
            name: 'deadly'
          },
          {
            title: '严重',
            name: 'seriousness'
          },
          {
            title: '一般',
            name: 'ordinary'
          },
          {
            title: '提示',
            name: 'energy'
          }
        ]
      },
      start: moment(),
      end: moment(),
      statisticalInfo: {
        deadly: 50,
        serious: 30,
        general: 10,
        hint: 10
      }
    };

    this.columns = [
      {
        id: 0,
        field: 'model',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.device.model'
        })
      },
      {
        id: 1,
        field: 'name',
        title: this.props.intl.formatMessage({ id: 'app.device.name' })
      },
      {
        id: 3,
        field: 'param',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.param' })
      },
      {
        id: 3,
        field: 'level',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.level' })
      },
      {
        id: 4,
        field: 'threshold',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.threshold'
        })
      },
      {
        id: 5,
        field: 'value',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.test.value'
        })
      },
      {
        id: 6,
        field: 'time',
        title: this.props.intl.formatMessage({ id: 'sysOperation.fault.time' })
      }
    ];
  }

  componentWillMount() {
    this.mounted = true;
    this.setState({ model: this.props.params.fault }, this.requestSearch);
    getChildDomainList(data => {
      this.mounted && this.initDomainList(data);
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidUpdate() {
    const { sidebarNode } = this.props;
    if (
      sidebarNode &&
      !sidebarNode.children &&
      sidebarNode.id != this.state.model
    ) {
      this.setState({ model: sidebarNode.id }, this.requestSearch);
    }
  }

  formatIntl = formatId => {
    const { intl } = this.props;
    return intl ? intl.formatMessage({ id: formatId }) : null;
  };

  initDomainList = data => {
    let domainList = Object.assign(
      {},
      this.state.domainList,
      { index: 0 },
      { value: data.length ? data[0].name : '' },
      { options: data }
    );
    this.setState({ domainList: domainList }, this.requestSearch);
  };

  requestSearch = () => {
    let model = '';
    switch (this.state.model) {
      case 'alarm':
        model = 'alerts';
        break;
      case 'fault':
        model = 'faults';
        break;
    }
    const { domainList, levelList, start, end, page } = this.state;
    const { pageSize, current } = page.toJS();
    if (domainList.options.length) {
      const domainId = domainList.options[domainList.index]['id'];
      // const level = levelList.options[levelList.index]['name'];
      const level = levelList.index;
      const offset = pageSize * (current - 1);
      const limit = pageSize;
      getFaultOrAlertList(
        model,
        domainId,
        level,
        start,
        end,
        offset,
        limit,
        res => {
          this.setState({ data: res });
        }
      );
    }
  };

  initPageSize = data => {
    let page = this.state.page.set('total', data.count);
    this.setState({
      page: page
    });
  };

  pageChange = (current, pageSize) => {
    let page = this.state.page.set('current', current);
    this.setState(
      {
        page: page
      },
      () => {
        this.requestSearch();
      }
    );
  };

  collapseHandler = id => {
    this.setState({
      [id]: !this.state[id]
    });
  };

  selectChange = (event, key) => {
    let index = event.target.selectedIndex;
    let list = this.state[key];
    list.index = index;
    list.value = list.options[index].name;
    this.setState({ [key]: list }, () => {
      this.requestSearch();
    });
  };

  dateChange = (id, value) => {
    this.setState({ [id]: value }, () => {
      const { start, end } = this.state;
      let prompt = start && end && end.isBefore(start);
      this.setState({
        prompt: Object.assign({}, this.state.prompt, { date: prompt })
      });
    });
  };

  drawChart = ref => {
    if (ref == null) {
      this.piechart.destroy();
    } else {
      const { statisticalInfo } = this.state;
      let data = [
        statisticalInfo.deadly,
        statisticalInfo.serious,
        statisticalInfo.general,
        statisticalInfo.hint
      ];
      this.piechart = new PieChart({
        wrapper: ref,
        data: data,
        color: ['#F83D59', '#FA919C', '#FA9E17', '#FBEF35']
      });
    }
  };

  render() {
    const {
      collapse,
      infoCollapse,
      page,
      data,
      domainList,
      typeList,
      levelList,
      start,
      end,
      statisticalInfo,
      allChecked,
      checked
    } = this.state;
    return (
      <Content className={'offset-right ' + (collapse ? 'collapsed' : '')}>
        <div className="heading">
          <Select
            id="domain"
            titleField={domainList.titleField}
            valueField={domainList.valueField}
            options={domainList.options}
            value={domainList.value}
            onChange={e => {
              this.selectChange(e, 'domainList');
            }}
          />
          {/* <Select id="type" titleField={typeList.titleField} valueField={typeList.valueField}
          options={typeList.options} value={typeList.value} onChange={(e) => {this.selectChange(e, 'typeList');}}/> */}
          <Select
            id="level"
            titleField={levelList.titleField}
            valueField={levelList.valueField}
            options={levelList.options}
            value={levelList.value}
            onChange={e => {
              this.selectChange(e, 'levelList');
            }}
          />
          <div className="datePicker">
            <DatePicker
              id="startDate"
              format="YYYY/MM/DD"
              placeholder="开始日期"
              style={{ width: '106px' }}
              defaultValue={start}
              value={start}
              onChange={value => this.dateChange('start', value)}
            />
            <span>-</span>
            <DatePicker
              id="endDate"
              format="YYYY/MM/DD"
              placeholder="结束日期"
              style={{ width: '106px' }}
              defaultValue={end}
              value={end}
              onChange={value => this.dateChange('end', value)}
            />
          </div>
        </div>
        <div className="table-container">
          <Table columns={this.columns} data={data} />
          <Page
            className={'page ' + (page.get('total') == 0 ? 'hidden' : '')}
            pageSize={page.get('pageSize')}
            current={page.get('current')}
            total={page.get('total')}
            onChange={this.pageChange}
          />
        </div>
        <SideBarInfo
          collapseHandler={this.collapseHandler}
          className={infoCollapse ? 'infoCollapse ' : ''}
        >
          <div className="panel panel-default statistical-info">
            <div
              className="panel-heading"
              role="presentation"
              onClick={() => {
                !collapse && this.collapseHandler('infoCollapse');
              }}
            >
              <span className="icon_chart" />
              {this.props.intl.formatMessage({
                id: 'sysOperation.statistical.info'
              })}
              <span className="icon icon_collapse pull-right" />
            </div>
            <div className={'panel-body' + (infoCollapse ? 'collapsed' : '')}>
              <div className="left">
                <div id="alarm" className="circle" ref={this.drawChart} />
              </div>
              <div className="right">
                <div className="count deadly">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.deadly')} :{' '}
                  {statisticalInfo.deadly
                    ? statisticalInfo.deadly
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count serious">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.serious')} :{' '}
                  {statisticalInfo.serious
                    ? statisticalInfo.serious
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count general">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.general')} :{' '}
                  {statisticalInfo.general
                    ? statisticalInfo.general
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count hint">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.hint')} :{' '}
                  {statisticalInfo.hint
                    ? statisticalInfo.hint
                    : this.formatIntl('sysOperation.noCount')}
                </div>
              </div>
            </div>
          </div>
        </SideBarInfo>
      </Content>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    sidebarNode: state.systemOperation.get('sidebarNode')
  };
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      treeViewInit,
      overlayerShow,
      overlayerHide
    },
    dispatch
  )
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(Fault));
