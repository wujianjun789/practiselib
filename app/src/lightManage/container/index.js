/**
 * Created by a on 2017/8/24.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/lightManage-map.less';
import '../../../public/styles/lightManage-control.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '.././../data/lightModel';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';
// import {sideBarToggled} from '../action/index'
class lightManageIndex extends Component {
  constructor(props) {
    super(props);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    getModelData(null, () => {
      this.mounted && this.initTreeData();
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidMount() {
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !== this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }

  initTreeData() {
    this.props.actions.treeViewInit(TreeData);
  }

  onToggle(node) {
    // console.log(node);
    // this.props.actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }
    return <div className={'container light-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName={'app.light'} router={this.props.router}/>
                    
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById
      // sideBarToggled: sideBarToggled
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(lightManageIndex);