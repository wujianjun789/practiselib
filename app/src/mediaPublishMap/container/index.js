/**
 * Created by a on 2017/10/19.
 */
import React,{Component} from 'react';

import Content from '../../components/Content';
import MapView from '../../components/MapView';
export default class MediaPublishMap extends Component{
    constructor(props){
        super(props);
        this.state = {
            map:{
                id: "mediaPublishMap",
                latlng: []
            }
        };
    }

    render(){
        const {map} = this.state;
        return <Content>
            <MapView mapData={map}></MapView>
        </Content>
    }
}