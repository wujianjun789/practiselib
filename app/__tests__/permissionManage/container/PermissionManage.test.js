import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../../src/store/configureStore';

import {PermissionManage} from '../../../src/permissionManage/container/index';
import Immutable from 'immutable';

const store = configureStore();
describe('<PermissionManage />',()=>{
    const store = configureStore();
    const columns = [
        {field:"username", title:"用户名称"},
        {field:"lastLoginDate", title:"最后登录时间"}
    ];
    const router = {push: jest.fn()};

    it('render normal', () =>{
        const root = mount(<Provider store={store}>
            <IntlProvider>
                <PermissionManage />
            </IntlProvider>
        </Provider>)
        const cmp = root.find('PermissionManage');
        const header = cmp.find('.heading');
        // const ins = cmp.instance();

        const search = root.state('search').toJS();
        const searchText = header.find('SearchText');
        expect(searchText.prop('placeholder')).toBe(search.placeholder);
        expect(searchText.prop('value')).toBe(search.value);

        const table = cmp.find('Table2');
        expect(table.prop('columns')).toEqual(ins.columns);
        expect(table.prop('data')).toEqual(root.state('datas'));

        const page = cmp.find('Page');
        const pageData = root.state('page');
        expect(page.prop('pageSize')).toBe(pageData.pageSize);
        expect(page.prop('current')).toBe(pageData.current);
        expect(page.prop('total')).toBe(pageData.total);
    });

    it('snapshot',() => {
        const cmp = renderer.create(<Provider store={store}>
            <IntlProvider>
                <PermissionManage />
            </IntlProvider>
        </Provider>);
        const tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    })
})