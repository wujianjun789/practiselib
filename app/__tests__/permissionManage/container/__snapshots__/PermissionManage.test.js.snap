exports[`<PermissionManage /> snapshot 1`] = `
<div
  className="container permission-manage">
  <div
    className="head">
    <div
      className="home"
      onClick={[Function]}>
      <span
        aria-hidden="true"
        className="glyphicon glyphicon-home" />
    </div>
    <span
      className="title">
      权限管理
    </span>
    <div
      className="user-center ">
      <svg
        className="svgOnload">
        <symbol
          id="alter">
          <path
            d="M100.5,3.5c-50.665,0-92.159,39.249-95.745,89h20.304C28.827,54.048,61.245,24,100.688,24 c41.974,0,76,34.026,76,76s-34.026,76-76,76c-16.36,0-31.498-5.186-43.894-13.982c2.125-3.098,6.972-11.393,7.296-11.393 c0.786,0,1.967-0.393,2.753-1.18l5.113-5.113l13.374-1.967c1.573-0.393,3.147-1.573,3.147-3.147l2.36-13.767l14.947-14.947 c11.8,5.113,25.96,2.36,35.401-6.687c12.587-12.587,12.587-33.041,0-45.627c-12.587-12.587-33.041-12.587-45.627,0 c-9.44,9.44-11.8,23.207-6.687,35.401l-38.941,38.94c-0.787,0.787-1.18,1.574-1.18,2.754s0.393,1.967,1.18,2.753L63.719,150.5 c0.308,0.308-5.236,8.111-7.214,11.314C39.692,149.765,27.988,131.038,25.3,109.5H5.017c5.001,48.32,45.841,86,95.483,86 c53.019,0,96-42.981,96-96S153.519,3.5,100.5,3.5z M119.551,69.99c5.507,0,9.833,4.327,9.833,9.833c0,5.507-4.327,9.833-9.833,9.833 c-5.507,0-9.833-4.327-9.833-9.833S114.044,69.99,119.551,69.99z M61.73,136.858l-2.753-2.753l27.534-27.534l2.753,2.753 L61.73,136.858z"
            fill="#8a8a8a" />
        </symbol>
        <symbol
          id="exit">
          <path
            d="M99.78,171.65c-39.805,0-71.732-32.341-71.732-71.732S60.39,28.187,99.78,28.187 c14.927,0,28.61,4.561,40.22,12.439c14.512,0,17.415,0.415,35.244,0.415C157.829,18.65,130.463,4.553,99.78,4.553 C46.707,4.553,4,47.26,4,100.333s42.707,95.78,95.78,95.78c30.683,0,57.634-14.098,75.463-36.488h-34.829 C128.805,167.089,114.707,171.65,99.78,171.65z M137.463,131.431l14.512,14.512L198,99.919l-46.024-46.024l-14.512,14.512 l19.902,19.902H73.195v23.22h84.171L137.463,131.431L137.463,131.431z"
            fill="#ff5c68" />
        </symbol>
      </svg>
      <div
        className="user-icon clearfix">
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-user" />
      </div>
      <ul
        className="user-list">
        <li
          onClick={[Function]}>
          <svg>
            <use
              height="200"
              transform="scale(0.08,0.08)"
              viewBox="0 0 20 20"
              width="200"
              x="0"
              xlinkHref="#alter"
              y="0" />
          </svg>
          <span>
            修改密码
          </span>
        </li>
        <li
          onClick={[Function]}>
          <svg>
            <use
              height="200"
              transform="scale(0.08,0.08)"
              viewBox="0 0 20 20"
              width="200"
              x="0"
              xlinkHref="#exit"
              y="0" />
          </svg>
          <span>
            退出管理系统
          </span>
        </li>
      </ul>
    </div>
    <div
      className="lanauage-switch ">
      <svg
        className="svgOnload">
        <symbol
          id="cn">
          <path
            d="M184.667,95v41.833l-2.333,7L175.5,148L24,148.167l-6.333-2.667l-3.479-5.833v-38.62c0,0,41.31,0.511,41.31,0.579 c-0.021,3.668,1.003,7.437,2.061,10.964C62.875,125.25,74,127.45,77.935,127.45c9.44,0,14.958-5.118,15.117-5.362 c0.113-0.209,0.199-0.433,0.255-0.664c0.024-0.272-0.017-4.189-0.051-4.494c-0.046-0.184-1.94-0.1-2.4,0.306 c-0.856,0.607-5.606,4.137-12.001,4.137c-9.23,0-15.781-9.04-15.781-19.07c0-18.302,11.176-21.734,15.372-21.734 C85.083,80,91.489,85.331,92.082,85.42c0.181,0.066,0.882-0.344,0.97-0.562c0.035-0.305,0.021-4.137,0-4.443 c-0.055-0.232-0.141-0.455-0.255-0.664c-0.205-0.3-4.922-5.158-14.095-5.158c-3.176-0.04-15.535,1.157-21.041,15.474 c-1.098,3.544-2.133,7.571-2.163,11.308c-0.001,0.159-41.31-0.531-41.31-0.531V63l4.145-6.5l7.354-3.188h6.297l-0.078-5.991h-6.318 c-9.872,0-17.874,8.003-17.874,17.874v71.498c0,9.872,8.003,17.874,17.874,17.874h148.103c9.872,0,17.874-8.003,17.874-17.874 V65.196l0,0c0-9.872-8.003-17.874-17.874-17.874H32l0.047,5.984l140.531-0.087l8.255,3.615l3.833,7.5v30.526l-41.364-0.218 l0.156,29.286c0.021,0.473-1.586,2.707-1.941,2.809c-0.356,0.051-4.582,0.082-5.209,0c-0.562-0.164-2.558-1.812-2.962-2.4 c-0.572-0.813-21.449-39.971-22.164-41.571l0,0c0,1.907,0,42.95,0,42.95c0.055,0.183-0.333,0.904-0.562,0.97h-5.464 c-0.227-0.07-0.617-0.787-0.562-0.97v-47.75c-0.091-0.871,2.192-2.97,2.911-2.962c0,0,4.701-0.079,5.362,0 c0.514,0.129,2.317,1.453,2.656,1.941c0.473,0.65,21.143,40.447,21.143,40.447l0,0c0-2.111,0-41.111,0-41.111 c-0.051-0.166,1.266-1.142,1.634-1.175h3.268c0.369,0.028,1.685,1.008,1.634,1.175l0.098,18.262L184.667,95z"
            fill="#8a8a8a" />
        </symbol>
        <symbol
          id="en">
          <path
            d="M56.425,101.469v22.758c-0.101,0.834,1.953,2.851,2.595,2.854h25.324c0.187,0.051,0.928-0.309,1.038-0.519 c0.142-0.293,0.222-4.179,0.104-4.812c-0.03-0.308-1.11-0.791-1.297-0.741H63.431v-18.785h17.54 c0.187,0.046,0.932-0.269,1.038-0.467c0.133-0.26,0.137-4.03,0-4.307c-0.111-0.208-0.852-0.575-1.038-0.519H63.171V80.585h20.757 c0.452,0.013,1.248-1.253,1.297-1.557c0.039-0.432,0.039-2.198,0-2.647c-0.049-0.321-0.84-1.637-1.297-1.609H59.02 c-0.641,0.003-2.696,2.02-2.595,2.854v23.686l-42.237,0.359V63l4.145-6.5l7.354-3.188h6.297l-0.078-5.991h-6.318 c-9.872,0-17.874,8.003-17.874,17.874v71.498c0,9.872,8.003,17.874,17.874,17.874h148.103c9.872,0,17.874-8.003,17.874-17.874 V65.196l0,0c0-9.872-8.003-17.874-17.874-17.874H32l0.047,5.984l140.531-0.087l8.255,3.615l3.833,7.5v30.526l-41.364-0.218 l0.156,29.286c0.021,0.473-1.586,2.707-1.941,2.809c-0.356,0.051-4.582,0.082-5.209,0c-0.562-0.164-2.558-1.812-2.962-2.4 c-0.572-0.813-21.449-39.971-22.164-41.571l0,0c0,1.907,0,42.95,0,42.95c0.055,0.183-0.333,0.904-0.562,0.97h-5.464 c-0.227-0.07-0.617-0.787-0.562-0.97v-47.75c-0.091-0.871,2.192-2.97,2.911-2.962c0,0,4.701-0.079,5.362,0 c0.514,0.129,2.317,1.453,2.656,1.941c0.473,0.65,21.143,40.447,21.143,40.447l0,0c0-2.111,0-41.111,0-41.111 c-0.051-0.166,1.266-1.142,1.634-1.175h3.268c0.369,0.028,1.685,1.008,1.634,1.175l0.098,18.262L184.667,95v41.833l-2.333,7 L175.5,148L24,148.167l-6.333-2.667l-3.479-5.833v-37.792L56.425,101.469z"
            fill="#8a8a8a" />
        </symbol>
      </svg>
      <div
        className="lanauage-icon">
        <svg>
          <use
            height="200"
            transform="scale(0.1957,0.195)"
            viewBox="0 0 36 21"
            width="200"
            x="-8"
            xlinkHref="#cn"
            y="-45" />
        </svg>
      </div>
      <ul
        className="list-group">
        <li
          className="list-group-item"
          onClick={[Function]}>
          <svg>
            <use
              height="200"
              transform="scale(0.1957,0.195)"
              viewBox="0 0 36 21"
              width="200"
              x="-8"
              xlinkHref="#cn"
              y="-45" />
          </svg>
          <span>
            简体中文
          </span>
        </li>
        <li
          className="list-group-item"
          onClick={[Function]}>
          <svg>
            <use
              height="200"
              transform="scale(0.1957,0.195)"
              viewBox="0 0 36 21"
              width="200"
              x="-8"
              xlinkHref="#en"
              y="-45" />
          </svg>
          <span>
            ENGLISH
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="content ">
    <div
      className="heading">
      <div
        className="searchText search"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="输入用户名称"
          type="search"
          value="" />
        <ul
          className="" />
        <span
          className="glyphicon glyphicon-search"
          onClick={[Function]} />
      </div>
      <button
        className="btn btn-primary"
        onClick={[Function]}>
        添加
      </button>
    </div>
    <div
      className="table-container">
      <div
        className="table-responsive ">
        <table
          className="table table-hover">
          <thead>
            <tr>
              <th
                className="hidden">
                <input
                  checked={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </th>
              <th>
                 
              </th>
              <th>
                用户名称
              </th>
              <th>
                最后登录时间
              </th>
              <th />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <ul
        className="rc-pagination page hidden"
        style={Object {}}
        unselectable="unselectable">
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="上一页">
          <a>
            
          </a>
        </li>
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="下一页">
          <a>
            
          </a>
        </li>
        <div
          className="rc-pagination-options" />
      </ul>
    </div>
  </div>
  <div
    className="overlayer "
    onClick={[Function]}>
    <div
      className=""
      id="animate" />
  </div>
</div>
`;
